include/seqan3/io/sequence_file/format_sam.hpp

```diff
@@ -0,0 +1,346 @@
+        size_t sequence_size_before = 0;
+        size_t sequence_size_after = 0;
+        if constexpr (!detail::decays_to_ignore_v<seq_type>)
+            sequence_size_before = ranges::size(sequence);
+
+        auto const is_tab = is_char<'\t'>;
+        auto const is_cntrl = is_char<'\n'>;
+        auto const is_free = is_char<'\r'>;
+        /* ID */
+        if (*stream_it != '@') //Ignore comment lines
+        {
+          // ID
+          if constexpr (!detail::decays_to_ignore_v<id_type>)
+          {
+            //TODO: Make truncate_ids also work if not all ids have a blank
+            if (options.truncate_ids)
+            {
+              ranges::copy(stream_view | view::take_until_or_throw(is_blank)
+                                       | view::char_to<value_type_t<id_type>>,
+                                   ranges::back_insert_iterator{id});
+              detail::consume(stream_view | ranges::view::take_while(is_tab));
+              detail::consume(stream_view | ranges::view::take_while(!is_tab));
+            }
+            else
+            {
+              ranges::copy(stream_view | view::take_until_or_throw(is_tab)
+                                       | view::char_to<value_type_t<id_type>>,
+                                   ranges::back_insert_iterator{id});
+            }
+
+          }
+          else
+          {
+            detail::consume(stream_view | view::take_line_or_throw);
+          }
+
+          // Jump over all parts we are not interested in
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+
+
+
+          auto seq_view = stream_view | ranges::view::remove_if(is_free || is_digit)   // ignore whitespace and numbers
```

Comment:
@h-2 I see that skipping whitespaces and numbers is also done when reading fasta/fastq so I am adressing you: Shouldn't the format rather throw an invalid-char error when a number occurrs whithin the sequence? 
@MitraDarja you commented to skip whitespaces but `is_free` checks (only) for `\r`? (I am also not sure if whitespaces should be allowed in SAM format)